buffer.go

```diff
@@ -49,7 +49,7 @@ func (b *buffer) fill(need int) error {
 	// grow buffer if necessary
 	// TODO: let the buffer shrink again at some point
 	//       Maybe keep the org buf slice and swap back?
-	if need > len(b.buf) {
+	if need > cap(b.buf) {
```

Comment:
Keep using `len` here.  `Read()` uses `len`, not `cap`.

Reply:
But the decision to scale the buffer should always `cap` based not `len` based, which is one of the reasons I explicitly called out the fact that for buf `len == cap`

I did have the code which expanded `len` as a second case but this would never hit due to the above.

Reply:
I know it. But no reason to use cap here.
Len is consistent with Read.